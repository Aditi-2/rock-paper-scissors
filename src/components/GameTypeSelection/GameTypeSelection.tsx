import React from "react";
import styled from "@emotion/styled";
import { GameType } from "../../common/types/game";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUser, faDesktop } from "@fortawesome/free-solid-svg-icons";

interface GameTypeSelectionProps {
  onGameSelection: (args: GameType) => void | Promise<void>
}

export const GameTypeSelection: React.FC<GameTypeSelectionProps> = ({onGameSelection}) => {
  return (
    <GameTypeSelectionStyled>
      <GameTypeStyled onClick={() => onGameSelection("PLAYER_VS_COMPUTER")}>
        <FontAwesomeIcon icon={faUser} className="player-icon" />
        <span className="verses">v/s</span>
        <FontAwesomeIcon icon={faDesktop} className="player-icon" />
      </GameTypeStyled>
      <GameTypeStyled onClick={() => onGameSelection("COMPUTER_VS_COMPUTER")}>
        <FontAwesomeIcon icon={faDesktop} className="player-icon" />
        <span className="verses">v/s</span>
        <FontAwesomeIcon icon={faDesktop} className="player-icon" />
      </GameTypeStyled>
    </GameTypeSelectionStyled>
  );
};

const GameTypeSelectionStyled = styled.div`
  padding: 2rem 0;
`;

const GameTypeStyled = styled.div`
  border: 1px solid #ece8e8;
  padding: 2rem;
  border-radius: 5px;
  margin: 1rem;
  display: flex;
  justify-content: center;
  @media only screen and (min-width: 800px) {
    width: 100%;
    max-width: 60%;
    margin: 3rem auto;
  }
  &:hover {
    background-color: #232a61;
  }
  & .verses {
    line-height: 7;
    padding: 0 4rem;
  }
  & .player-icon {
    font-size: 70px;
    margin: 0 5px;

    @media only screen and (min-width: 800px) {
      font-size: 125px;
    }
  }
`;
