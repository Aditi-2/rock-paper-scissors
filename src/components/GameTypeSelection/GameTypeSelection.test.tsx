import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { create } from "react-test-renderer";
import { GameTypeSelection } from "./GameTypeSelection";

const onGameSelect = jest.fn();

const getComponent = () => <GameTypeSelection onGameSelection={onGameSelect} />;

test("GameTypeSelection Component snapshot", () => {
  const component = create(getComponent()).toJSON();
  expect(component).toMatchSnapshot();
});

test("renders GameTypeSelection Component", () => {
  render(getComponent());
  const allVs = screen.getAllByText(/v\/s/i);
  allVs.forEach((vs) => expect(vs).toBeInTheDocument());
});

test("GameTypeSelection buttons register click", () => {
  render(getComponent());
  const images = screen.getAllByRole("img", { hidden: true });
  images.forEach((image) => fireEvent.click(image));
  expect(onGameSelect).toHaveBeenCalledTimes(images.length);
});
