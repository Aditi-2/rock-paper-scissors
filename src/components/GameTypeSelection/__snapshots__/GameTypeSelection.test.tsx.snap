// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameTypeSelection Component snapshot 1`] = `
<div
  className="css-7gsmpx"
>
  <div
    className="css-1v9wwo0"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-user fa-w-14 player-icon"
      data-icon="user"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="verses"
    >
      v/s
    </span>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-desktop fa-w-18 player-icon"
      data-icon="desktop"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528 0H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h192l-16 48h-72c-13.3 0-24 10.7-24 24s10.7 24 24 24h272c13.3 0 24-10.7 24-24s-10.7-24-24-24h-72l-16-48h192c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zm-16 352H64V64h448v288z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="css-1v9wwo0"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-desktop fa-w-18 player-icon"
      data-icon="desktop"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528 0H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h192l-16 48h-72c-13.3 0-24 10.7-24 24s10.7 24 24 24h272c13.3 0 24-10.7 24-24s-10.7-24-24-24h-72l-16-48h192c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zm-16 352H64V64h448v288z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="verses"
    >
      v/s
    </span>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-desktop fa-w-18 player-icon"
      data-icon="desktop"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528 0H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h192l-16 48h-72c-13.3 0-24 10.7-24 24s10.7 24 24 24h272c13.3 0 24-10.7 24-24s-10.7-24-24-24h-72l-16-48h192c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zm-16 352H64V64h448v288z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
